sql/moz-fx-data-shared-prod/mozilla_vpn_derived/guardian_apple_events_v1/query.sql

```diff
@@ -99,21 +108,31 @@ CROSS JOIN
         CAST(cancellation_reason AS INT64) AS revocation_reason,
       FROM
         UNNEST(
-          ARRAY_CONCAT(
-            COALESCE(legacy_subscriptions.apple_receipt.receipt.in_app, []),
-            COALESCE(legacy_subscriptions.apple_receipt.latest_receipt_info, [])
-          )
+          [
+            -- in order by priority
+            STRUCT(
+              "latest_receipt_info" AS source,
+              legacy_subscriptions.apple_receipt.latest_receipt_info AS receipts
+            ),
+            STRUCT(
+              "receipt.in_app" AS source,
+              legacy_subscriptions.apple_receipt.receipt.in_app AS receipts
+            )
+          ]
         )
-        WITH OFFSET AS _offset
+        WITH OFFSET AS _source_offset
+      CROSS JOIN
+        UNNEST(COALESCE(receipts, []))
+        WITH OFFSET AS _receipts_offset
       WHERE
         latest_original_transaction_id = original_transaction_id
       QUALIFY
         1 = ROW_NUMBER() OVER (
           PARTITION BY
             is_trial_period
           ORDER BY
-            expires_date_ms DESC,
-            _offset DESC
+            _source_offset,
+            _receipts_offset
```

Comment:
I agree with prioritizing `latest_receipt_info` records over `receipt.in_app` records, and I'm pretty sure there will only be one `latest_receipt_info` record per `original_transaction_id`.  However, there can be multiple `receipt.in_app` records per `original_transaction_id` and they don't appear to be in a consistent order, so I'd suggest still sorting secondarily by the expires date instead of the receipts offset.
```suggestion
            expires_date_ms DESC
```